code/datums/components/storage/concrete/rped.dm

```diff
@@ -15,6 +15,39 @@
 			M.balloon_alert(M, "The RPED can only hold machine parts!")
 		return FALSE
 
+/datum/component/storage/concrete/rped/quick_empty(mob/M)
+	var/atom/A = parent
+	if(!M.canUseStorage() || !A.Adjacent(M) || M.incapacitated())
+		return
+	if(check_locked(null, M, TRUE))
+		return FALSE
+	A.add_fingerprint(M)
+	var/list/things = contents()
+	var/lowest_rating = INFINITY
+	for(var/obj/item/B in things)
+		if(istype(B, /obj/item/stock_parts/cell))
+			var/obj/item/stock_parts/cell/C = B
+			if(C.rating < lowest_rating)
+				lowest_rating = C.rating
+		else if(B.get_part_rating() < lowest_rating)
+			lowest_rating = B.get_part_rating()
+	for(var/obj/item/B in things)
+		if(istype(B, /obj/item/stock_parts/cell))
+			var/obj/item/stock_parts/cell/C = B
+			if(C.rating > lowest_rating)
+				things.Remove(B)
+		else if(B.get_part_rating() > lowest_rating)
+			things.Remove(B)
+	if(lowest_rating == INFINITY)
+		to_chat(M, "<span class='notice'>There's no parts to dump out from [parent].</span>")
+		return
+	to_chat(M, "<span class='notice'>You start dumping out Tier/Cell rating [lowest_rating] parts from [parent].</span>")
+	var/turf/T = get_turf(A)
+	var/datum/progressbar/progress = new(M, length(things), T)
+	while (do_after(M, 10, TRUE, T, FALSE, CALLBACK(src, .proc/mass_remove_from_storage, T, things, progress)))
+		stoplag(1)
```

Comment:
why

Reply:
> why

I don't know, that's just how it looks on Citadel, which is the original codebase I ported this from.

Reply:
evil

Reply:
Right, so looking at it on Citadel, their version of a do_after looks advanced as hell, there's an entire file just dedicated to the thing compared to how our version works.

Reply:
Huh, this is more or less how it's done here:
https://github.com/BeeStation/BeeStation-Hornet/blob/c3de34b93be036bb468a5b8056529c3974623250/code/datums/components/storage/storage.dm#L268